src/components/button/index.tsx

```diff
@@ -0,0 +1,14 @@
+import type { QwikIntrinsicElements } from "@builder.io/qwik";
+import { component$, Slot } from "@builder.io/qwik";
+
+import styles from "./button.module.css";
+
+export const Button = component$(
+  ({ class: classProp, ...props }: QwikIntrinsicElements["button"]) => {
+    return (
+      <button class={`${styles["button"]} ${classProp}`} {...props}>
```

Comment:
I pull out the `class` prop because we want to apply our own style, but also allow for other times when whoever is rendering a `Button` wants to give it their own styles. This way, you can do something like
```tsx
<Button class={styles["my-button-class"]}>Hello!</Button>
```
and the class will both 1. be rendered and 2. not overwrite our default styles.